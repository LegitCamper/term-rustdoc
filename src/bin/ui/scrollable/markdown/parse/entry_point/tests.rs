use super::{markdown_iter, parse};
use insta::{assert_debug_snapshot as snap, assert_display_snapshot as shot};

#[test]
fn parse_markdown() {
    let doc = r#"
# h1 `code`

aaa b *c* d **e**. ~xxx~ z.

1 *c **ss** d sadsad xxx* `yyyy`

```
let a = 1;
```

> rrr sss 
> tt

- [x] done!
    - nested list
- [ ] undone
    1. *a*
    2. `b`
"#;
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    # h1 `code`

    aaa b c d e. xxx z.

    1 c ss d sadsad xxx `yyyy`

    let a = 1;

    rrr sss tt

    * [x] done!
      * nested list
    * [ ] undone
      1. a
      2. `b`

    "###);
}

#[test]
fn parse_markdown_links() {
    let doc = "
[a](b), [c], [e][c].

[c]: d

[`f`][c].

## h2 [c] [`h`][c]
";
    snap!(markdown_iter(doc).collect::<Vec<_>>());
    shot!(parse(doc), @r###"
    a, c, e.

    `f`.

    ## h2 c `h`

    "###);
}
