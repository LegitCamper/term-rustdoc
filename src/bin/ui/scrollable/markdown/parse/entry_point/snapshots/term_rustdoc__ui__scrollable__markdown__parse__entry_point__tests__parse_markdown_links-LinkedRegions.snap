---
source: src/bin/ui/scrollable/markdown/parse/entry_point/tests.rs
expression: regions
---
LinkedRegions {
    tag: {
        Heading(
            0,
        ): TargetRegion {
            targets: [
                SelectedRegion {
                    row_start: 14,
                    row_end: 15,
                    col_start: 0,
                    col_end: 2,
                },
            ],
        },
    },
}
